MachObfuscator/Mach/ObjC/ObjcSymbols.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  ObjcSymbols.swift
+//  MachObfuscator
+//
+
+import Foundation
+
+/// These protocols provide architecture independent and quite rich interface
+/// for accessing ObjC metadata.
+
+protocol ObjcIvar {
+    var name: StringInData { get }
+    var type: StringInData { get }
+}
+
+protocol ObjcProperty {
+    var name: StringInData { get }
+    var attributes: StringInData { get }
+}
+
+extension ObjcProperty {
```

Comment:
This could probably be moved to `Mach+properties.swift` and made `private`.

Reply:
`isDynamic` is one of the attributes of the property, not only a random extension to implement something, so I think it fits here. It makes underlying metadata easier to analyse.  Currently it is indeed used only in one place in Mach+properties, but IMO is sufficiently generic to be available everywhere.

Reply:
> sufficiently generic to be available everywhere

This way we could end up removing `private` keywords all across the codebase ;)

For me this seems like a kind of pre-mature generalisation. We will end-up with method visible anywhere, just in single place. I think we should extract this when it becomes in fact needed somewhere else.

Reply:
OK, moved. But how do you discover that a private extension doing something interesting already exists (IDE does not help, because private extension in not visible in other files)? Do you have to know entire code base or look everywhere? Or does one not care, and implement the same functionality in private extensions several times?

Reply:
Ok, I see your point. My motivation was to hide it as a private extension so the code could be more-refactorable. I won't insist if you want to change it to be `internal`.

Reply:
Lets leave it the way you suggest. I was only wondering how do you manage such private extensions (BTW, I discovered some useful extensions - like `fileOffset` - in Obfuscator by chance: by browsing code not related with what I was implementing)

Reply:
Not all extensions are private. As for `isDynamic` – `Mach+properties` could do `filter`ing without the `extension Property`. Extracting the `isDynamic` in a `private extension` seems to be just better-formatted internal implementation. I think it is good to extract as much code, as possible. But is it a good idea to make everything non-private? Regarding this particular property, now after this conversation... I am not sure :D

How do I manage such internal implementations? – I think I don't have such a guideline. I have a lot of `private extension`s in my apps, the things just works. I think it is an intuition.

What do you do in your apps? Do you make everything non-private? Or do you have any golden-rule?